Explan more about externalizing config (#36)
OpenLiberty/guide-microprofile-config-profile: README.adoc

Deploying the application to a Kubernetes environment using the Open Liberty Operator is an optional step in this guide.

To further explore deploying microservices using Kubernetes and the Open Liberty Operator, you can read the following guides:

* https://openliberty.io/guides/openliberty-operator-intro.html[Deploying a microservice to Kubernetes using Open Liberty Operator^]
* https://openliberty.io/guides/openliberty-operator-openshift.html[Deploying a microservice to OpenShift 4 using Open Liberty Operator^]

> Ensuring security in the production environment is essential. In the previous sections, you have learnt how to externalize credentials and other properties for accessing the `system` service by using MicroProfile Config, making the application more adaptable to different environments without changing code or configuration files. In the this section, you'll learn how to make use of Kubernetes secrets to provide the credentials and how to pass them to the `query` service through the MicroProfile Config.

Reviewer: ```suggestion
A secure production environment is essential to application security. In the previous sections, you learned how to externalize credentials and other properties for accessing the `system` service by using MicroProfile Config. This strategy makes the application more adaptable to different environments without changing code or configuration files. 

In the this section, you'll learn how to use Kubernetes secrets to provide the credentials and how to pass them to the `query` service by using MicroProfile Config.
```

> Ensuring security in the production environment is essential. In the previous sections, you have learnt how to externalize the credentials and other properties for accessing the `system` service by using the MicroProfile Config that makes the application more adaptable to different environments without changing code or configuiration files.  In the this section, you'll learn how to make use of Kubernetes secret to provide the credentials and how to pass them to the `query` service through the MicroProfile Config.

Reviewer 2: ```suggestion
Ensuring security in the production environment is essential. In the previous sections, you have learnt how to externalize credentials and other properties for accessing the `system` service by using MicroProfile Config, making the application more adaptable to different environments without changing code or configuration files. In the this section, you'll learn how to make use of Kubernetes secrets to provide the credentials and how to pass them to the `query` service through the MicroProfile Config.
```

I think "In the previous sections, you have learnt how to externalize the credentials and other properties for accessing the `system` service by using the MicroProfile Config that makes the application more adaptable to different environments without changing code or configuiration files." is too lengthy? so I split it with a comma. 

Also, 'configuration' was spelled incorrectly in the original sentence.

        --from-literal password=[password]
----

> To learn more how to manage the secrets, visit the https://kubernetes.io/docs/tasks/configmap-secret/managing-secret-using-kubectl[Managing Secrets using kubectl^] documentation.

Reviewer: ```suggestion
For more information about  managing secrets, see the https://kubernetes.io/docs/tasks/configmap-secret/managing-secret-using-kubectl[Managing Secrets using kubectl^] documentation.
```

Finally, write up the [hotspot file=0]`deploy.yaml` deployment file to configure the deployment of the [hotspot=system file=0]`system` and [hotspot=query file=0]`query` microservices by using the Open Liberty Operator. The [hotspot=sys-app-credentials-1 hotspot=sys-app-credentials-2 hotspot=sys-app-credentials-3 hotspot=sys-app-credentials-4 file=0]`sys-app-credentials` Kubernetes secrets set the environment variables [hotspot=default.username file=0]`DEFAULT_USERNAME` and [hotspot=default.password file=0]`DEFAULT_PASSWORD` for the `system` microservice, and [hotspot=system.user file=0]`SYSTEM_USER` and [hotspot=system.password file=0]`SYSTEM_PASSWORD` for the `query` microservice.

// file 5
deploy.yaml
[source, XML, linenums, role='code_column']
----
include::finish/deploy.yaml[]
----

> If you want to override the other property, you can specify it through the `env` in the `deploy.yaml`. For example, set the [hotspot=context.root file=5]`CONTEXT_ROOT` environment variable in the `system` deployment and [hotspot=system.contextRoot file=5]`SYSTEM_CONTEXTROOT` environment variable in the `query` deployment.

Reviewer 2: ```suggestion
If you want to override another property, you can specify it through the `env` sections in the `deploy.yaml`. For example, set the [hotspot=context.root file=5]`CONTEXT_ROOT` environment variable in the `system` deployment and the [hotspot=system.contextRoot file=5]`SYSTEM_CONTEXTROOT` environment variable in the `query` deployment.
```

> If you want to override another property, you can specify it through the `env` sections in the `deploy.yaml`. For example, set the [hotspot=context.root file=5]`CONTEXT_ROOT` environment variable in the `system` deployment and the [hotspot=system.contextRoot file=5]`SYSTEM_CONTEXTROOT` environment variable in the `query` deployment.

Reviewer: ```suggestion
If you want to override another property, you can specify it in the `env` sections of the `deploy.yaml` file. For example, set the [hotspot=context.root file=5]`CONTEXT_ROOT` environment variable in the `system` deployment and the [hotspot=system.contextRoot file=5]`SYSTEM_CONTEXTROOT` environment variable in the `query` deployment.
```

> If you want to override another property, you can specify it through the `env` sections in the `deploy.yaml`. For example, set the [hotspot=context.root file=5]`CONTEXT_ROOT` environment variable in the `system` deployment and the [hotspot=system.contextRoot file=5]`SYSTEM_CONTEXTROOT` environment variable in the `query` deployment.

Reviewer: > set the `CONTEXT_ROOT` environment variable in the `system` deployment and the `SYSTEM_CONTEXTROOT` environment variable in the `query` deployment.

Is this right? seems confusing that the SYSTEM_CONTEXTROOT variable goes with the `query` instead of the `system`

Author: It is right. In the system service perspective, CONTEXT_ROOT is for its own context root.  In the query service perspective, it needs to know the context root of the system service, so SYSTEM_CONTEXTROOT.
